### ISSUE TITLE
Local installation impossible in 3.5.25

### ISSUE THREAD

**Issue by Michael1972** [contao/core#8681]
when performing a local installation in mampp, the following error-message appears after confirming the license-aggreement: 

```
Fatal error: Uncaught exception Error with message Class 'Database\' not found thrown in system/modules/core/library/Contao/Database.php on line 165
#0 system/modules/core/library/Contao/Controller.php(1078): Contao\Database::getInstance()
#1 system/modules/core/library/Contao/Template.php(298): Contao\Controller::setCorsHeaders()
#2 system/modules/core/classes/BackendTemplate.php(120): Contao\Template->output()
#3 system/modules/core/controllers/BackendInstall.php(882): Contao\BackendTemplate->output()
#4 system/modules/core/controllers/BackendInstall.php(133): Contao\BackendInstall->outputAndExit()
#5 contao/install.php(20): Contao\BackendInstall->run()
#6 {main}
```

**Comment by fritzmg** [comment]
The problem seems to be, that in your environment, the following code does not return `null`:
```
\Environment::get('httpOrigin');
```
This causes `Controller::setCorsHeaders` (see [Controller.php#L1065](https://github.com/contao/core/blob/3.5.25/system/modules/core/library/Contao/Controller.php#L1065)) to initiate the `Database` instance (see [Controller.php#L1078](https://github.com/contao/core/blob/3.5.25/system/modules/core/library/Contao/Controller.php#L1078)), which of course will fail, since the config is incomplete.

Try changing [this line](https://github.com/contao/core/blob/3.5.25/system/modules/core/library/Contao/Controller.php#L1068) to:
```php
if (!$strOrigin || \Environment::get('url') == $strOrigin)
```

**Comment by fritzmg** [comment]
@Toflar according to https://wiki.mozilla.org/Security/Origin#When_Origin_is_served_.28and_when_it_is_.22null.22.29 POST requests are also supposed to set the `HTTP_ORIGIN` header (though I cannot reproduce that locally with neither Chrome nor Firefox).

When accepting the license, you are sending a POST request. Thus
```php
\Environment::get('httpOrigin');
```
will not be `null` and `Controller::setCorsHeaders` will be executed which will initiate the `Database` - which will fail.

**Comment by Toflar** [comment]
Yeah this was fixed in Contao 4 already, only needs to be backported.

**Comment by Michael1972** [comment]
Works with the changed controller.php #1065. Great. Though I don't understand the code and its conclusions, the problem obviously doesn't appear with firefox or crome. I'm working with safari. Thanks.

**Comment by leofeyer** [comment]
Fixed in 813ac5661996afc7d12bcb5856a728e5e73cabf9.
